frontend/lib/Repository/rest_client.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:adpv_frontend/Repository/UserRepository/auth_gateway.dart';
+import 'package:adpv_frontend/Repository/UserRepository/user_gateway.dart';
+import 'package:dio/dio.dart';
+
+class RestClient {
+  final Dio _client = Dio();
+  final UserGateway _userGateway = UserGateway();
+
+  Future<Response<T>> get<T>(
+      String path, {
+        Map<String, dynamic>? queryParameters,
+      }) =>
+      _performQuery(
+            () => _client.get(
+          path,
+          queryParameters: queryParameters,
+        ),
+      );
+
+  Future<Response<T>> delete<T>(
+      String path, {
+        Map<String, dynamic>? queryParameters,
+      }) =>
+      _performQuery(
+            () => _client.delete(
+          path,
+          queryParameters: queryParameters,
+        ),
+      );
+
+  Future<Response<T>> post<T>(
+      String path,
+      {
+        Map<String, dynamic>? queryParameters,
+        dynamic data,
+      }) =>
+      _performQuery(
+            () => _client.post(
+          path,
+          queryParameters: queryParameters,
+          data: data,
+        ),
+      );
+
+  Future<Response<T>> put<T>(
+      String path, {
+        Map<String, dynamic>? queryParameters,
+        dynamic data,
+      }) =>
+      _performQuery(
+            () => _client.post(
+          path,
+          queryParameters: queryParameters,
+          data: data,
+        ),
+      );
+
+  Future<T> _performQuery<T>(Function requestMethod) async {
+    final AuthResponse authResponse = await _userGateway.getFromMemory();
+
+    if (authResponse.success) {
+      final String token = authResponse.tokens!.accessToken;
+      _client.options.headers["accept"] = "*/*";
+      _client.options.headers["Authorization"] = "Bearer $token";
+      try {
+        return await requestMethod();
+      } on DioError catch (error) {
+        return Future.error(error);
+      }
+    }
+
+    return Future.value(null);
```

Comment:
[Sugestia] Tu bym się zastanowił, czy jednak może nie lepiej by było zwracać jakiś error, niż `Future.value(null)`, jak pierwotnie sugerowałem

Reply:
To jest dokładny copy-paste Twojej sugestii, imo jest ok, przynajmniej na ten moment

Reply:
Yup, wiem że copy-paste mojego, dlatego piszę, że nie wiem, czy dobrze to zasugerowałem. Jak uważasz, że jest okej, to można zostawić, albo co najwyżej zmienić na `Future.error(null)`, żeby zasygnalizować, że to nie jest porządana przez nas zwrotka